src/components/Contacts/ContactDetails/ContactDontationsTab/PartnershipInfo/EditPartnershipInfoModal/EditPartnershipInfoModal.tsx

```diff
@@ -0,0 +1,431 @@
+  const [
+    updateContactPartnership,
+    { loading: updating },
+  ] = useUpdateContactPartnershipMutation();
+  const pledgeCurrencies = constants?.pledgeCurrencies;
+  const method = contact.lastDonation
+    ? contact.lastDonation.paymentMethod
+    : t('None');
+
+  const lastGift = contact.lastDonation
+    ? `${currencyFormat(
+        contact.lastDonation.amount.amount,
+        contact.lastDonation.amount.currency,
+      )}`
+    : '';
+
+  const contactPartnershipSchema: yup.SchemaOf<
+    Pick<
+      ContactUpdateInput,
+      | 'id'
+      | 'status'
+      | 'pledgeAmount'
+      | 'pledgeFrequency'
+      | 'pledgeCurrency'
+      | 'pledgeReceived'
+      | 'pledgeStartDate'
+      | 'nextAsk'
+      | 'noAppeals'
+    >
+  > = yup.object({
+    id: yup.string().required(),
+    status: yup.mixed<StatusEnum>().oneOf(Object.values(StatusEnum)).nullable(),
+    pledgeAmount: yup.number().moreThan(-1).nullable(),
+    pledgeStartDate: yup.string().nullable(),
+    pledgeReceived: yup.boolean().default(false).nullable(),
+    pledgeCurrency: yup.string().nullable(),
+    nextAsk: yup.string().nullable(),
+    noAppeals: yup.boolean().default(false).nullable(),
+    pledgeFrequency: yup.mixed<PledgeFrequencyEnum>().nullable(),
+  });
+
+  const onSubmit = async (
+    attributes: Pick<
+      ContactUpdateInput,
+      | 'id'
+      | 'status'
+      | 'pledgeAmount'
+      | 'pledgeFrequency'
+      | 'pledgeCurrency'
+      | 'pledgeReceived'
+      | 'pledgeStartDate'
+      | 'nextAsk'
+      | 'noAppeals'
+    >,
+  ) => {
+    console.log(attributes.nextAsk);
+    await updateContactPartnership({
+      variables: {
+        accountListId: accountListId ?? '',
+        attributes,
+      },
+    });
+
+    enqueueSnackbar(t('Partnership information updated successfully.'), {
+      variant: 'success',
+    });
+    handleClose();
+  };
+
+  const updateStatus = (
+    status: StatusEnum,
+    setFieldValue: (name: string, value: StatusEnum | number | null) => void,
+  ) => {
+    if (status !== StatusEnum.PartnerFinancial) {
+      setFieldValue('status', status);
+      setFieldValue('pledgeAmount', 0);
+      setFieldValue('pledgeFrequency', null);
+    } else {
+      setFieldValue('status', status);
+    }
+  };
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      title={t('Edit Partnership')}
+      handleClose={handleClose}
+    >
+      <Formik
+        initialValues={{
+          id: contact.id,
+          status: contact.status,
+          pledgeAmount: contact.pledgeAmount,
+          pledgeFrequency: contact.pledgeFrequency,
+          pledgeReceived: contact.pledgeReceived,
+          pledgeCurrency: contact.pledgeCurrency,
+          pledgeStartDate: contact.pledgeStartDate,
+          nextAsk: contact.nextAsk,
+          noAppeals: contact.noAppeals,
+        }}
+        validationSchema={contactPartnershipSchema}
+        onSubmit={onSubmit}
+      >
+        {({
+          values: {
+            status,
```

Comment:
If we added the rest of the contact, all those other values would be accessible here. But we don't need them since only certain values are going to be edited by this modal. The best practice I've discovered is to only pass in what is going to be edited in the form.